modules/news/news.service.js

```diff
@@ -26,22 +28,35 @@ class NewsService {
     throw new Error(NEWS_NOT_FOUND);
   }
 
-  async updateNews(id, news) {
+  async updateNews(id, news, upload) {
     const foundNews = await News.findById(id);
     if (!foundNews) {
       throw new Error(NEWS_NOT_FOUND);
     }
 
+    if (upload[0] && upload[1]) {
+      news.author.image = await uploadLargeImage(upload[0]);
+      news.image = await uploadLargeImage(upload[1]);
+    } else if (upload[0]) {
+      news.author.image = await uploadLargeImage(upload[0]);
+    } else if (upload[1]) {
+      news.image = await uploadLargeImage(upload[1]);
+    }
+
     if (await this.checkNewsExist(news, id)) {
       throw new Error(NEWS_ALREADY_EXIST);
     }
     return await News.findByIdAndUpdate(id, news, { new: true });
   }
 
-  async addNews(data) {
+  async addNews(data, upload) {
     if (await this.checkNewsExist(data)) {
       throw new Error(NEWS_ALREADY_EXIST);
     }
+
+    data.author.image = await uploadLargeImage(upload[0]);
```

Comment:
You can check if you have something in upload. If there are no files you can throw error.